### PR TITLE
Mostly visual changes

### PR DESCRIPTION
**Pull request by or-else**
Visual changes + all formatters moved into a single file.

One thing that does not work right: if a reply is to a message not currently loaded into the messages-view, clicking it will never work, even if the message is later loaded.

Replying to messages with attachments is also broken. I'll fix it tomorrow.

### CODE REVIEW

File: src/lib/formatters.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        attr.onClick = this.onImagePreview;
+        attr.className += ' image-clickable';
+      }
+      attr.loading = 'lazy';
+    } else {
+      attr.src = 'img/broken_image.png';
+    }
+  } else {
+    // Use custom element instead of <img>.
+    el = UploadingImage;
```
Comment (aforge) [inline src/lib/formatters.js:69]: Please, import `UploadingImage`.

File: src/views/messages-view.jsx [hunk 2]
```diff
@@ -959,14 +975,12 @@ class MessagesView extends React.Component {
...
-    if (element) {
-      element.scrollIntoView({block: "center", behavior: "smooth"});
-      element.style.backgroundColor = 'rgb(0, 0, 0, 0.4)';
-      setTimeout(() => { element.style.backgroundColor = ''; } , 1000);
+    if (ref && ref.current) {
+      ref.current.scrollIntoView({block: "center", behavior: "smooth"});
+      // ref.current.style.backgroundColor = 'rgb(0, 0, 0, 0.4)';
+      ref.current.classList.add('flash');
+      setTimeout(() => { ref.current.classList.remove('flash') } , 1000);
+      //setTimeout(() => { ref.current.style.backgroundColor = ''; } , 1000);
```
[the remark this answers is not in the document]
Comment (or-else) [reply src/views/messages-view.jsx:983]: done

File: src/lib/formatters.js (hunk 1 above)
Comment (or-else) [reply src/lib/formatters.js:69]: done

**Comment by or-else** [comment]
The attachments work as expected now.

**Comment by aforge** [comment]
Can you check large image uploading please?
The client appears to drop the api key param on the request.

**Comment by aforge** [comment]
> Can you check large image uploading please?
> The client appears to drop the api key param on the request.

To be more specific: after uploading, fetching the image locally in order to render it produces a `403: valid API key required`.

The request URL is `http://.../v0/file/s/<file>.jpe` - while we expect `http://.../v0/file/s/<file>.jpe?apikey=...&other-params...`

**Comment by or-else** [comment]
> The client appears to drop the api key param on the request.

This is an interesting bug. The JS method assignments don't work the way I expect it to work:
https://jsfiddle.net/Ldmy4b1h/1/
`this` is lost in assignment.

**Pull request merged**